### PR TITLE
TMA-509: Provisioning creates zombie clients when project creation limit reached

### PR DESCRIPTION
**Pull request by vhtien**
## Todos
- [ ] Tests
- [ ] Documentation
- [ ] Changelog
- [ ] Build passing

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-rubocop-zuul/815/ : SUCCESS in 14s
- https://ci.intgdc.com/job/gooddata-ruby-tests-zuul/771/ : FAILURE in 2m 40s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/543/ : FAILURE in 25s

File: lib/gooddata/lcm/actions/provision_clients.rb [hunk 1]
```diff
@@ -38,28 +39,37 @@ def call(params)
...
-                end
-              }
-            end
+                }
+              end
 
-            tmp
+              tmp
+            end
+          rescue => e
```
Comment (panjan) [inline lib/gooddata/lcm/actions/provision_clients.rb]: Can we write a regression test for this?

File: lib/gooddata/lcm/actions/provision_clients.rb [hunk 2]
```diff
@@ -38,28 +39,37 @@ def call(params)
...
+            params.gdc_logger.error "Problem occurs when provisioning clients. Purge all invalid clients now ..."
+            res = PurgeClients.send(:call, params)
```
Comment (vhtien) [inline lib/gooddata/lcm/actions/provision_clients.rb]: It's just a way to invoke a function. I am not allowed to write this or something?

File: lib/gooddata/lcm/actions/provision_clients.rb (hunk 1 above)
Comment (vhtien) [inline lib/gooddata/lcm/actions/provision_clients.rb]: Honestly I don't know how to break the provisioning brick by reaching the creation limit. Could you suggest me how to do it? Without using mock, please!

File: lib/gooddata/lcm/actions/provision_clients.rb (hunk 2 above)
Comment (panjan) [inline lib/gooddata/lcm/actions/provision_clients.rb]: Using `send` for ordinary method calls is quite non-standard. However, it's a minor issue I am willing to live with.

File: lib/gooddata/lcm/actions/provision_clients.rb (hunk 1 above)
Comment (korczis) [inline lib/gooddata/lcm/actions/provision_clients.rb]: @vhtien You have to use mock for this. There is no other option. Many times we agreed that all bugfixes should be covered with test. At least twice @zsvoboda confirmed that. If you have some good reason for not covering this with test then ask @sitina for confirmation of exception (not having a test for this) in TMA room on hipchat. Thanks in advance.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/9/ : SUCCESS in 19s
- https://ci.intgdc.com/job/gooddata-ruby-rubocop-zuul/843/ : SUCCESS in 14s

File: lib/gooddata/lcm/actions/provision_clients.rb (hunk 1 above)
Comment (vhtien) [inline lib/gooddata/lcm/actions/provision_clients.rb]: done

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-merge-compliance-check/2/ : SUCCESS in 11s
- https://ci.intgdc.com/job/gooddata-ruby-rubocop-zuul/847/ : SUCCESS in 13s
- https://ci.intgdc.com/job/gooddata-ruby-tests-zuul/798/ : SUCCESS in 45m 46s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/570/ : UNSTABLE in 13m 55s

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-merge-compliance-check/6/ : CODE REVIEW VIOLATED in 6s
- gooddata-ruby-rubocop-zuul : SKIPPED
- gooddata-ruby-unit-tests-zuul : SKIPPED
- gooddata-ruby-integration-tests-zuul : SKIPPED
- gooddata-ruby-lcm-integration-tests-zuul : SKIPPED

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-merge-compliance-check/11/ : SUCCESS in 11s
- https://ci.intgdc.com/job/gooddata-ruby-rubocop-zuul/867/ : SUCCESS in 14s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/14/ : SUCCESS in 3m 11s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/816/ : SUCCESS in 44m 29s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/589/ : UNSTABLE in 19m 27s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-merge-compliance-check/13/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-ruby-rubocop-zuul/869/ : SUCCESS in 24s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/16/ : SUCCESS in 3m 01s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/818/ : SUCCESS in 42m 48s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/591/ : SUCCESS in 21m 20s

**Pull request merged**